src/redux/reducers/rootReducer.js

```diff
@@ -0,0 +1,122 @@
+  },
+  trans: {
+    lastUpdated: null,
+    isFetching: false,
+    isSynced: false,
+    trans: []
+  }
+}
+
+export function masterReducer(state = initialState, action) {
+  switch (action.type) {
+    case AuthActions.SIGNUP_REQUEST:
+    case AuthActions.SIGNUP_FAILURE:
+    case AuthActions.SIGNUP_SUCCESS:
+    case AuthActions.LOGIN_REQUEST:
+    case AuthActions.LOGIN_FAILURE:
+    case AuthActions.LOGIN_SUCCESS:
+    case AuthActions.LOGOUT_REQUEST:
+    case AuthActions.LOGOUT_FAILURE:
+    case AuthActions.LOGOUT_SUCCESS:
+    case AuthActions.GET_USER_INFO_REQUEST:
+    case AuthActions.GET_USER_INFO_FAILURE:
+    case AuthActions.GET_USER_INFO_SUCCESS:
+      return {
+        ...state,
+        user: authReducer(state.user, action)
+      };
+    case TripActions.GET_TRIPS_REQUEST:
+    case TripActions.GET_TRIPS_FAILURE:
+    case TripActions.GET_TRIPS_SUCCESS:
+    case TripActions.ADD_TRIP_REQUEST:
+    case TripActions.ADD_TRIP_FAILURE:
+    case TripActions.ADD_TRIP_SUCCESS:
+    case TripActions.UPDATE_TRIP_REQUEST:
+    case TripActions.UPDATE_TRIP_FAILURE:
+    case TripActions.UPDATE_TRIP_SUCCESS:
+    case TripActions.DELETE_TRIP_REQUEST:
+    case TripActions.DELETE_TRIP_FAILURE:
+    case TripActions.DELETE_TRIP_SUCCESS:
+    case TripActions.GET_TRIP_INFO_REQUEST:
+    case TripActions.GET_TRIP_INFO_FAILURE:
+    case TripActions.GET_TRIP_INFO_SUCCESS:
+      return {
+        ...state,
+        trips: tripReducer(state.trips, action)
+      };
+    case PlanActions.GET_PLANS_REQUEST:
+    case PlanActions.GET_PLANS_FAILURE:
+    case PlanActions.GET_PLANS_SUCCESS:
+    case PlanActions.ADD_PLAN_REQUEST:
+    case PlanActions.ADD_PLAN_FAILURE:
+    case PlanActions.ADD_PLAN_SUCCESS:
+    case PlanActions.UPDATE_PLAN_REQUEST:
+    case PlanActions.UPDATE_PLAN_FAILURE:
+    case PlanActions.UPDATE_PLAN_SUCCESS:
+    case PlanActions.DELETE_PLAN_REQUEST:
+    case PlanActions.DELETE_PLAN_FAILURE:
+    case PlanActions.DELETE_PLAN_SUCCESS:
+    case PlanActions.GET_PLAN_INFO_REQUEST:
+    case PlanActions.GET_PLAN_INFO_FAILURE:
+    case PlanActions.GET_PLAN_INFO_SUCCESS:
+      return {
+        ...state,
+        plans: planReducer(state.plans, action)
+      };
+    case TranActions.GET_TRANS_REQUEST:
+    case TranActions.GET_TRANS_FAILURE:
+    case TranActions.GET_TRANS_SUCCESS:
+    case TranActions.ADD_TRAN_REQUEST:
+    case TranActions.ADD_TRAN_FAILURE:
+    case TranActions.ADD_TRAN_SUCCESS:
+    case TranActions.UPDATE_TRAN_REQUEST:
+    case TranActions.UPDATE_TRAN_FAILURE:
+    case TranActions.UPDATE_TRAN_SUCCESS:
+    case TranActions.DELETE_TRAN_REQUEST:
+    case TranActions.DELETE_TRAN_FAILURE:
+    case TranActions.DELETE_TRAN_SUCCESS:
+    case TranActions.GET_TRAN_INFO_REQUEST:
+    case TranActions.GET_TRAN_INFO_FAILURE:
+    case TranActions.GET_TRAN_INFO_SUCCESS:
+      return {
+        ...state,
+        trans: tranReducer(state.trans, action)
+      };
+    default:
+      return state;
+  }
+}
```

Comment:
I'm pretty sure you don't need a massive switch/case like this; that's what combineReducers does when you pass it all of the different reducers. It makes separate branches based on the reducers' names, and each reducer is called for every single action (but they will only ever actually do things when the action corresponds to something in the switch/case statement.

Reply:
I'll look into combineReducers again then. I was a little confused the last time I took a look at it, and I saw this approach somewhere as well. With my next pull request that contains all of the Redux changes I'll have our masterReducer improved.

Reply:
Cool! Sounds good.